src/shims/unix/foreign_items.rs

```diff
@@ -936,7 +925,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     this.write_null(dest)?;
                 } else {
                     this.write_null(&result)?;
-                    this.write_scalar(this.eval_libc("ERANGE"), dest)?;
+                    this.write_scalar(LibcError("ERANGE"), dest)?;
```

Comment:
The issue here is that `From<IoError>` is not implemented for `rustc_const_eval::interpret::Scalar<machine::Provenance>`. This can probably be handled in another commit.

Reply:
Leave this unchanged, it doesn't set errno.